From d49ce24ef2979bb2ef485a918d525bc295ba777a Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Tue, 16 Dec 2014 22:22:15 -0500
Subject: [scsi] libcxgbi: fix freeing skb prematurely

Message-id: <1418768535-2637-8-git-send-email-svemuri@redhat.com>
Patchwork-id: 102112
O-Subject: [RHEL7.1 PATCH BZ 1174982 7/7] libcxgbi: fix freeing skb prematurely
Bugzilla: 1174982
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

Upstream : https://git.kernel.org/cgit/linux/kernel/git/davem/net-next.git/commit/?id=ed481a33ee824bfee20319fc478503926bcf5f56

With debug turned on the debug print would access the skb after it is freed.
Fix it to free the skb after the debug print.

Signed-off-by: Karen Xie <kxie@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit ed481a33ee824bfee20319fc478503926bcf5f56)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/cxgbi/libcxgbi.c b/drivers/scsi/cxgbi/libcxgbi.c
index 541d811..0c23553 100644
--- a/drivers/scsi/cxgbi/libcxgbi.c
+++ b/drivers/scsi/cxgbi/libcxgbi.c
@@ -2294,10 +2294,12 @@ int cxgbi_conn_xmit_pdu(struct iscsi_task *task)
   return err;
  }
 
- kfree_skb(skb);
  log_debug(1 << CXGBI_DBG_ISCSI | 1 << CXGBI_DBG_PDU_TX,
   "itt 0x%x, skb 0x%p, len %u/%u, xmit err %d.\n",
   task->itt, skb, skb->len, skb->data_len, err);
+
+ kfree_skb(skb);
+
  iscsi_conn_printk(KERN_ERR, task->conn, "xmit err %d.\n", err);
  iscsi_conn_failure(task->conn, ISCSI_ERR_XMIT_FAILED);
  return err;
-- 
1.7.1